### PR TITLE
option to suppress warnings for key mismatch on service object (margueritepd)

### PR DESCRIPTION
**Pull request by BartoszBlizniak**
Raising this on behalf of @margueritepd in order to let the tests run with correct values.

## Context
We don't want to store credentials in state. The cloudsmith provider stores the service account's key in state, but continues to work even if the key in the state doesn't match reality, in which case it issues a warning. To get around storing the key in state, we will have a workflow where, after the service account is provisioned, we will rotate the API key via the UI. The only problem with this is that warnings will be issued in future TF plans and applies which we have no intention of addressing.

## This change

This change adds the option to suppress warnings about the key in state not matching reality.

## Testing

I compiled this locally and added it to a local terraform plugin dir and ran some tests.

On unmodified terraform code to provision a service account, where the key in state matches reality:
Terraform wants to add a new attribute: warn_on_key_difference = true.
Applying will set this attribute in the state.

On unmodified terraform code to provision a service account, where the key in state does NOT match reality:
The warning is issued, and the plan returns with Terraform wanting to add the new attribute: warn_on_key_difference = true
if I update the terraform code so that warn_on_key_difference = false and plan, the warning is still issued, and terraform wants to add the new attribute: warn_on_key_difference = false
After applying, the warning is no longer issued on subsequent terraform plans.

## Alternatives considered

Ideally we would be able to turn off storing the key in state completely. That would remove the need to rotate the key in the UI afterward. But that seemed like a more invasive change which I was not confident executing.
I tried adding lifecycle.ignore_changes to the resource but since the key is a computed attribute and not one specified in the terraform code, this did nothing.

## Feedback i'm looking for

Let me know if you want me to try another approach, or if you want me to do further testing!

### CODE REVIEW

**Comment by BartoszBlizniak** [comment]
Hey @cloudsmith-io/engineering - could I get some eyes on the PR and approval, ty!

File: cloudsmith/resource_service.go [hunk 1]
```diff
@@ -272,6 +275,12 @@ func resourceService() *schema.Resource {
 				Computed:    true,
 				Sensitive:   true,
 			},
+			"warn_on_key_difference": {
+				Type:        schema.TypeBool,
+				Description: "Whether to warn if service's current key differs from key in state.",
+				Optional:    true,
+				Default:     true,
+			},
```
Comment (chrisimcevoy) [inline cloudsmith/resource_service.go:283]: Feels a bit weird to make this an "opt-in" which defaults to true. I'd be more inclined to make it an "opt-out" which defaults to false.

File: cloudsmith/resource_service.go
Comment (chrisimcevoy) [inline cloudsmith/resource_service.go:1]: Should there be a service.md docs change that goes along with this?

**Review by chrisimcevoy — COMMENTED** [review]
Still considering this approach in general. In the meantime, have some comments.

File: cloudsmith/resource_service.go (hunk 1 above)
Comment (BartoszBlizniak) [reply cloudsmith/resource_service.go:283]: I think either will be interchangeable, I'd personally be inclined to leave this as is, easy change anyway.

File: cloudsmith/resource_service.go
Comment (BartoszBlizniak) [reply cloudsmith/resource_service.go:1]: Yup, adding those now

**Comment by BartoszBlizniak** [comment]
Closing this PR in favour of #63

**Pull request closed without merging**